src/main/java/Duke/TaskManager.java

```diff
@@ -0,0 +1,152 @@
+package Duke;
+
+import Tasks.Deadline;
+import Tasks.Event;
+import Tasks.Task;
+import Tasks.Todo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class TaskManager {
+    private static ArrayList<Task> tasks = new ArrayList<>();
+    private static int numTask = 0;
+
+    public static final String TASK_ADDED_MESSAGE = "The task has been added: \n";
+    public static final String NO_SUCH_TASK_MESSAGE = "There is no task by that number \n";
+    public static final String TASK_COMPLETE_MESSAGE = "Congrats on finishing a task! Have a cookie!\n";
+    public static final String NO_TASKS_MESSAGE = "No Tasks\n";
+    public static final String TASK_DELETED_MESSAGE = "The following task has been deleted:\n";
+
+    public TaskManager() {
+    }
+
+    /**
+     * This section deals with file reading and writing
+     */
+    public static void processStoredTasks(String taskString) {
+        String[] taskDetails = taskString.split("\\|", 4);
+        boolean isDone = taskDetails[1].trim().equals("X");
+        switch (taskDetails[0].trim()) {
+        case "T":
+            tasks.add(new Todo(taskDetails[2].trim(), isDone));
+            numTask++;
+            break;
+        case "D":
+            String[] dueDate = taskDetails[3].split("by:");
+            tasks.add(new Deadline(taskDetails[2].trim(), isDone, dueDate[1].trim()));
+            numTask++;
+            break;
+        case "E" :
+            String[] startDate = taskDetails[3].split("at:");
+            tasks.add(new Event(taskDetails[2].trim(), isDone, startDate[1].trim()));
+            numTask++;
+            break;
+        default:
+            break;
+        }
+    }
+
+    public static void readFile() throws FileNotFoundException {
+        File taskFile = new File("data/tasks.txt");
+        Scanner scanner = new Scanner(taskFile);
+        if (taskFile.exists() && !taskFile.isDirectory()) {
+            while (scanner.hasNext()) {
+                processStoredTasks(scanner.nextLine());
+            }
+        } else {
```

Comment:
You can make the [happy path](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#make-the-happy-path-prominent) more prominent here. If the condition is not met, you can return the function call immediately. Then you can avoid nesting your subsequent logic